weaviate/collections/batch/batch_wrapper.py

```diff
@@ -44,32 +53,35 @@ def wait_for_vector_indexing(
         if shards is not None and not isinstance(shards[0], Shard):
             raise TypeError(f"'shards' must be of type List[Shard]. Given type: {type(shards)}.")
 
-        def is_ready(how_many: int) -> bool:
+        async def is_ready(how_many: int) -> bool:
             try:
-                return all(
-                    all(self._get_shards_readiness(shard))
-                    for shard in shards or self._batch_data.imported_shards
+                readinesses = await asyncio.gather(
+                    *[
+                        self.__get_shards_readiness(shard)
+                        for shard in shards or self._batch_data.imported_shards
+                    ]
                 )
+                return all(all(readiness) for readiness in readinesses)
             except Exception as e:
                 print(
                     f"Error while getting class shards statuses: {e}, trying again with 2**n={2**how_many}s exponential backoff with n={how_many}"
                 )
                 if how_many_failures == how_many:
                     raise e
                 time.sleep(2**how_many)
-                return is_ready(how_many + 1)
+                return await is_ready(how_many + 1)
 
         count = 0
-        while not is_ready(0):
+        while not self._event_loop.run_until_complete(is_ready, count):
```

Comment:
Because `ConnectionV4` only contains `async` methods now using `httpx.AsyncClient` and `grpclib.Channel` so rather than persist a `sync` version of `ConnectionV4`, all the functionality above it has to use the root `async` implementation either directly in an `async` method or wrapped by the `_EventLoop` code

An added benefit here of having `async def is_ready` is that the `__get_shards_readiness` calls can be ran concurrently, using `asyncio.gather`, rather than serially as it was before